[new module] Add tRNAScan-SE module

prototaxites opened the pull request:
Adds module for tRNAScan-SE. 

In local testing, I was having a problem where when running in a Singularity container, the default internal /tmp space would fill up very quickly and the program would fail due to lack of disk space. I implemented a solution that moves the working directory from /tmp to the nextflow work dir, but this requires modifying an internal configuration file. Open to better solutions, but this one was working for me.

## PR checklist

Closes #XXX <!-- If this PR fixes an issue, please link it here! -->

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

SPPearce reviewed (commented):
Looks generally good.
Can you add some ontologies please, it'd be good to have those included for new modules.

prototaxites commented:
Thanks for the review! Added ontologies as requested, and also removed the input val parameters that add flags to write optional output files, as suggested to me in https://github.com/nf-core/modules/pull/8026 - these can be specified by the pipeline developer in ext.args instead.

prototaxites commented:
@SPPearce Does this look OK to you now?

SPPearce commented on modules/nf-core/trnascanse/main.nf:
```diff
@@ -0,0 +1,75 @@
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args      = task.ext.args   ?: ''
+    def prefix    = task.ext.prefix ?: "${meta.id}"
+    def input     = fasta.toString() - ~/\.gz$/
+    def unzip     = fasta.getExtension() == "gz" ? "gunzip -c ${fasta} > ${input}" : ""
+    def cleanup   = fasta.getExtension() == "gz" ? "rm ${input}" : ""
```
```suggestion
    def args    = task.ext.args   ?: ''
    def prefix  = task.ext.prefix ?: "${meta.id}"
    def input   = fasta.toString() - ~/\.gz$/
    def unzip   = fasta.getExtension() == "gz" ? "gunzip -c ${fasta} > ${input}" : ""
    def cleanup = fasta.getExtension() == "gz" ? "rm ${input}" : ""
```

SPPearce commented on modules/nf-core/trnascanse/main.nf:
```diff
@@ -0,0 +1,75 @@
+    """
+
+    stub:
+    def prefix      = task.ext.prefix ?: "${meta.id}"
```
```suggestion
    def prefix = task.ext.prefix ?: "${meta.id}"
```

SPPearce reviewed (commented):
Is it likely that people are going to want to pass a custom trnascan.conf file directly to the process?

prototaxites commented:
I think it's unlikely - it's not a very well-documented feature of the tool. I only found it as I was trying to fix the issue with Singularity!

SPPearce commented:
> I think it's unlikely - it's not a very well-documented feature of the tool. I only found it as I was trying to fix the issue with Singularity!

Ok, fair enough.
